adapters/gamma/gamma.go

```diff
@@ -0,0 +1,245 @@
+		}
+	}
+	headers := http.Header{}
+	headers.Add("Accept", "*/*")
+	headers.Add("x-openrtb-version", "2.5")
+	if request.Device != nil {
+		addHeaderIfNonEmpty(headers, "User-Agent", request.Device.UA)
+		addHeaderIfNonEmpty(headers, "X-Forwarded-For", request.Device.IP)
+		addHeaderIfNonEmpty(headers, "Accept-Language", request.Device.Language)
+		if request.Device.DNT != nil {
+			addHeaderIfNonEmpty(headers, "DNT", strconv.Itoa(int(*request.Device.DNT)))
+		}
+	}
+	headers.Add("Connection", "keep-alive")
+	headers.Add("cache-control", "no-cache")
+	headers.Add("Accept-Encoding", "gzip, deflate")
+
+	return &adapters.RequestData{
+		Method:  "GET",
+		Uri:     thisURI,
+		Headers: headers,
+	}, errors
+}
+func (a *GammaAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+	if len(request.Imp) == 0 {
+		err := &errortypes.BadInput{
+			Message: "No impressions in the bid request",
+		}
+		errs = append(errs, err)
+		return nil, errs
+	}
+	var invalidImpIndex = make([]int, 0, 0)
+
+	for i := 0; i < len(request.Imp); i++ {
+		if request.Imp[i].Banner != nil {
+			bannerCopy := *request.Imp[i].Banner
+			if bannerCopy.W == nil && bannerCopy.H == nil && len(bannerCopy.Format) > 0 {
+				firstFormat := bannerCopy.Format[0]
+				bannerCopy.W = &(firstFormat.W)
+				bannerCopy.H = &(firstFormat.H)
+			}
+			request.Imp[i].Banner = &bannerCopy
+		} else if request.Imp[i].Video == nil {
+			err := &errortypes.BadInput{
+				Message: fmt.Sprintf("Gamma only supports banner and video media types. Ignoring imp id=%s", request.Imp[i].ID),
+			}
+			errs = append(errs, err)
+			invalidImpIndex = append(invalidImpIndex, i)
+		}
+	}
+
+	var adapterRequests []*adapters.RequestData
+	if len(invalidImpIndex) == 0 {
+		for _, imp := range request.Imp {
+			adapterReq, errors := a.makeRequest(request, imp)
+			if adapterReq != nil {
+				adapterRequests = append(adapterRequests, adapterReq)
+			}
+			errs = append(errs, errors...)
+		}
+	} else if len(request.Imp) == len(invalidImpIndex) {
+		//only true if every Imp was not a Banner or a Video
+		err := &errortypes.BadInput{
+			Message: fmt.Sprintf("No valid impression in the bid request"),
+		}
+		errs = append(errs, err)
+		return nil, errs
+	} else {
+		var j int = 0
+		for i := 0; i < len(request.Imp); i++ {
+			if j < len(invalidImpIndex) && i == invalidImpIndex[j] {
+				j++
+			} else {
+				adapterReq, errors := a.makeRequest(request, request.Imp[i])
+				if adapterReq != nil {
+					adapterRequests = append(adapterRequests, adapterReq)
+				}
+				errs = append(errs, errors...)
+			}
+		}
+	}
+
+	return adapterRequests, errs
+}
+
+func (a *GammaAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+
```

Comment:
In case of `http.StatusNoContent`, I'd suggest returning `nil, nil` because it's not a `BadServerResponse` if your adapter is sending back an empty response for some reason.